### DISCUSSION TITLE
Ignore submodule on deploy

### DISCUSSION THREAD

**Discussion by iantearle** [deployphp/deployer#2436] [Help needed] [answered]
Is it possible to ignore git submodules on deploy? Have a folder of themes as a submodule, but don't want it to be synced on deploy of the main repo

**Comment by iantearle** [comment, marked as answer]
To answer my self and anyone else looking;

`set('git_recursive', false);`

Docs needs some help. Will maybe get round to helping out.

**Reply by antonmedv** [reply]
Yes, working on docs will help.

**Reply by mrhackcz** [reply]
@antonmedv Hi! Is there an alternative in the 7.x version? I'm trying to deploy a repository with a submodule (I want a submodule), but so far, I'm not succeeding. Thank you

**Reply by antonmedv** [reply]
Create a separate task to install submodules?

**Reply by mrhackcz** [reply]
Yep, I rewrote update_code a little bit. It's for an outdated project, so I hope it will be removed soon. I was just curious if there is already an option for that. Thanks!

**Reply by AWqxKAWERbXo** [reply]
Is it possible to share your result with me? @mrhackcz

**Reply by mrhackcz** [reply]
> Is it possible to share your result with me? @mrhackcz

Sure, it's nothing special. But it works.
https://gist.github.com/mrhackcz/b3074f97f4ede2d03eed9c4de892b034

**Reply by AWqxKAWERbXo** [reply]
Thank you so much @mrhackcz  

With your hint, I managed to copy the original `deploy:update_code` task and modified it to overwrite it with your code in lines 46-50.

If there's already a if/else-statement that checks for `update_code_strategy`, then it would be so easy to add a third value beside `archive` and `clone`. E.g. like `clone_submodule`.
